### ISSUE TITLE
Cluster breaks on push and collecting logs

### ISSUE THREAD

**Issue by dottodot** [flynn/flynn#3900]
Using the latest version on  Ubuntu 16

Just tried to push an update and cluster went down and all apps reported service unavailable. Oddly though the update did get deployed.

Tried `flynn-host fix` which didn't work on the first node but did on the second. Then tried to `collect-debug-info` and it crashed again and had to do `flynn-host fix` to recover.

I'm now reluctant to do anything in fear of bringing it all down again.

**Comment by dottodot** [comment]
One thing I've just found in the blobstore log is

> 2017-02-06T11:16:53.018680Z app[web.cluster2-5ecf9d82-6580-4bc7-9da8-4a3f84b7441b]: t=2017-02-06T11:16:53+0000 lvl=eror msg="heartbeat failed" component=discoverd service=blobstore addr=100.100.96.5:80 err="Error sending HTTP request, errors: service_unavailable: discoverd: no known leader,service_unavailable: discoverd: no known leader,service_unavailable: discoverd: no known leader"

**Comment by lmars** [comment]
@dottodot you should check the available disk space / RAM on your servers (`df -h` and `free -m`).

**Comment by dottodot** [comment]
This is my disk space which I think is fine 

```
Filesystem                                               Size  Used Avail Use% Mounted on
udev                                                     994M     0  994M   0% /dev
tmpfs                                                    201M   11M  190M   6% /run
/dev/vda1                                                 39G  5.4G   34G  14% /
tmpfs                                                   1001M     0 1001M   0% /dev/shm
tmpfs                                                    5.0M     0  5.0M   0% /run/lock
tmpfs                                                   1001M     0 1001M   0% /sys/fs/cgroup
overlay                                                   39G  5.4G   34G  14% /tmp/tmp.hldgaaWEwL/mnt
overlay                                                   39G  5.4G   34G  14% /tmp/tmp.jEsKvGVulY/mnt
flynn-default                                            7.5G     0  7.5G   0% /flynn-default
flynn-default/data                                       7.5G     0  7.5G   0% /flynn-default/data
flynn-default/squashfs                                   7.5G     0  7.5G   0% /flynn-default/squashfs
flynn-default/ext2                                       7.5G     0  7.5G   0% /flynn-default/ext2
/dev/zd0                                                 2.0M  2.0M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7
/dev/zd16                                                3.9M  3.9M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/34f7be5eb9e26c96d79d88f562f0c9eeda3ec7ffc4f902b5281118f95dfc446e
/dev/zd32                                                9.3M  9.3M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/23139a7706aca8d4f06fac6bba6545c741318bb62af59f601f12a0aa683be2f9
/dev/zd48                                                 63M   63M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c
/dev/zd64                                                434M  434M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/4c1cc7b8a1a83a1ecc0cde8ab6ec0e41b8e718d6f964486244952d99d329f201
/dev/zd80                                                 18M   18M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/880bc70da660da92a5a0bb5f65f60596279fc27d549316d83caa58a003154ba9
/dev/zd96                                                 33M   33M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/53ac780c79b5d20722c0839bceba2ed4109d9004fe025ac7dee13d9df84b3f95
/dev/zd112                                               109M  109M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/50bdd0dc04d964158a4fef3b3eb303e25e1f0e6cbed41baea02cf32ca8d0cfc9
/dev/zd128                                                11M   11M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/d7ac4709c0ee9bc4c56d0c1e9bd9fd235d6941f2ce0d679e725db0b2702c6ef9
/dev/zd144                                               112M  112M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/f1239bded7c110c0d84c019bb424d557777cf3ebd1433a7a9c883bdb5bafef47
/dev/zd160                                               2.8M  2.8M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/7ad90f2a91113ae82af2ff59f101f8ecc4cf2b7d7eadee50c33c45a1831800dd
/dev/zd176                                                42M   42M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/ba6fee0ad0edcd61145efd9c1ed761e22a9c8815e8bc2b75ca52511c211550bd
/dev/zd192                                               128K  128K     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/18962e0ce642ebcdb47b50e63c94d439141ce86a77d9d923e90ea0fd3cfc13e5
/dev/zd208                                                40M   40M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/e9963f6a065c8db0753f88df39d16284382169cd94aad81adfe42c833138b049
/dev/zd224                                               6.9M  6.9M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/71d691c20e1a1f39f9e1e4c46e8d210e1735c71916d0e94de5e6fa8184c3b09a
/dev/zd240                                               2.5M  2.5M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/d31be30310bb8f2122e9a973005baa9f9454f9f47a0b04ce278d60105a518301
/dev/zd256                                                86M   86M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/4290e7093802596c51aa8160daf7b2220f4cd1199467262216fa510f8753c530
/dev/zd272                                                45M   45M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/50e451a986f01cd4641f55dd390f574c6b63967b6e955c75cb7afbf2a87a84cd
/dev/zd288                                                94M   94M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/0dae691254b5d389f18f768b9fb785676f07639c6ce4884c3b0e79b0893c97b9
/dev/zd304                                               5.3M  5.3M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/6a204e6ae5b472126abbb3eb1bf25d6be8b0a782ce6e79f4fcb89d7c259bc8f8
/dev/zd320                                                30M   30M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/bf9fc7cf2f371748cddf016b206072207d5230530165be0e3b3d8b089c463669
/dev/zd336                                               3.2M  3.2M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/e6e8dac7e9de7b044af61ca960421cb9d065fb99aa0d0305b2b1153536d6798d
/dev/zd352                                               2.5M  2.5M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/807964592874224ead2c92754ed9b1ea640b881c7d3dcb837f4964cb3a2127a0
/dev/zd368                                                79M   79M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/3beb8f6b6a10004f0dd906174e60eedc0bd3ecd1edb00688f5b8b1a18ed6b3d4
flynn-default/data/6b592a4c-17dd-42f8-aa53-d709ef28ad7d  7.5G  640K  7.5G   1% /var/lib/flynn/volumes/zfs/mnt/data/6b592a4c-17dd-42f8-aa53-d709ef28ad7d
/dev/zd384                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-ca8b4b68-3380-4330-bdee-9668bfb77ba1
/dev/zd400                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-41f923b6-c21c-4339-a33f-ac4e5f279d31
flynn-default/data/e9fc9537-a1b1-46a3-bfdc-d8395debfa03  7.7G  228M  7.5G   3% /var/lib/flynn/volumes/zfs/mnt/data/e9fc9537-a1b1-46a3-bfdc-d8395debfa03
/dev/zd416                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-a4aa49cb-90a7-4703-b86e-5d552b6c4036
/dev/zd432                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-abcd5fad-fe4a-4296-8f86-5e9f975d321d
/dev/zd448                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-acfa044c-0d00-4e9d-b29a-097d57c0f02d
/dev/zd464                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-7ec107fc-1383-4356-a922-0bf1ace4a5de
/dev/zd480                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-9022a514-9550-473b-8dca-7aa1a7800200
/dev/zd496                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-13580e9c-8458-4123-9ee3-54a3d64d2525
/dev/zd512                                                27M   27M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/cc9c21d968cb944960eaa81356fa32f2f6943a397f12c4e011277a1319b8f8a8
/dev/zd544                                                42M   42M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/a767199255e5c34f1c783200a5e938d602dae33562138c14682de429c9ec2430
/dev/zd560                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-3195d00f-0450-434c-ad99-5f2df83ac91e
/dev/zd576                                               114M  114M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/30968586af9be860e21de903b85c23b109d3fe544ee2dd5f6ce6466966c8f834
/dev/zd640                                               139M  139M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/badeb06c53460ebae34b1bc2e6fcd70be248616ccca83b351d77d84f40b7ab91
/dev/zd656                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-921d2188-aebd-4929-bc65-6374e92f5aed
/dev/zd672                                                97M  2.2M   95M   3% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-38e64423-a952-4276-9d17-09161e247a1c
/dev/zd688                                               111M  111M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/9d7a0be394505e1dc8fe4da0c6e15cb39fa3a2b1d923b27f713b0c705c203a83
/dev/zd704                                                97M  2.2M   95M   3% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-e4115e3c-8802-44c9-b08f-93149bdda233
/dev/zd720                                                97M  2.2M   95M   3% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-4aa34636-de7e-4170-9277-65b480921e47
/dev/zd528                                                31M   31M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/51d9e91cd8ab121572eb7f62eddd197b9ad2445461c35d1b0c773d81de3f38d3
/dev/zd736                                                38M   38M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/5c10946778eed1a128da946a2d1f38a77a18ca9121f00038dd6ba46c59460d27
/dev/zd752                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-7ae2b738-90e6-4386-8e3a-ec19d7e605ae
/dev/zd768                                               114M  114M     0 100% /var/lib/flynn/volumes/zfs/mnt/squashfs/e5ab205195d1dbf3a1ae50b531f0f1ca9456a05bc1397f511e4c67245c10e2a4
/dev/zd784                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-8cd0dd35-26be-4692-998a-976b1765391b
/dev/zd800                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-bee188d4-71b4-4d85-9bf9-4b3be2bd2ceb
tmpfs                                                    201M     0  201M   0% /run/user/0
/dev/zd592                                                97M  1.6M   96M   2% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-403206ba-f05a-49c8-bf75-e1852ab51101
/dev/zd608                                                97M   12M   86M  13% /var/lib/flynn/volumes/zfs/mnt/ext2/cluster1-9749f59e-1194-4d89-8afa-905fcde4efe3
```
and this is the memory use which is nearly maxed out, but it's odd that it's not been a issue before
```
              total        used        free      shared  buff/cache   available
Mem:           2000        1168         161          36         670         364
Swap:          4095         436        3659
```

**Comment by lmars** [comment]
@dottodot if you can't collect the debug info, create a tarball of `/var/log/flynn` and send it to logs@flynn.io.

**Comment by dottodot** [comment]
OK they've been sent to you.

**Comment by dottodot** [comment]
Did you managed to find anything that would help solve this?

**Comment by dottodot** [comment]
Not sure if it's related but this was the first time I'd tried Ubuntu 16. Just created a new cluster on Ubuntu 14 and seem to be working ok.

**Comment by dottodot** [comment]
New cluster is the same, doing a git push basically breaks it and once broken can't collect debug info and making my cluster unusable, now completely stuck and not able to push and updates.

**Comment by titanous** [comment]
Can you connect to the servers and check what is consuming CPU?

**Comment by dottodot** [comment]
gitreceived is using the most seen it hit 95%

**Comment by dottodot** [comment]
and npm over 100%

**Comment by titanous** [comment]
NPM can hammer the filesystem pretty hard, that might be what's going on.

**Comment by titanous** [comment]
Check `iotop`.

**Comment by dottodot** [comment]
If I scale the app down to one before pushing it seem to work ok.

My other concern is weather my cluster is now working correctly, as I can't collect debug info and what other issues has it caused.

The same thing happened yesterday on the previous cluster and half through the night one of the server crashed completely and rebooted.

**Comment by dottodot** [comment]
Actually it not ok been stuck creating release for 11 minutes now

**Comment by dottodot** [comment]
and now stuck in a release that I can't get rid of

**Comment by kenaniah** [comment]
Experiencing the same exact issue on v20170116.0, but with plenty of disk space, memory, and CPU. 3-node cluster.

```
# sudo zpool list
NAME            SIZE  ALLOC   FREE  EXPANDSZ   FRAG    CAP  DEDUP  HEALTH  ALTROOT
flynn-default  80.0G  8.38G  71.6G         -    18%    10%  1.00x  ONLINE  -

# top
cpu1 0.5% 
cpu2 1%
load average 0.05 0.23 0.25

# free -m 
              total        used        free      shared  buff/cache   available
Mem:           7982        1634         157          63        6190        5490
Swap:             0           0           0
```

**Comment by titanous** [comment]
@kenaniah Please collect the logs and provide the error that you saw.

**Comment by kenaniah** [comment]
@titanous tried `flynn-host collect-debug-info` and got the following:

```
INFO[02-10|21:40:35] uploading logs and debug information to a private, anonymous gist 
INFO[02-10|21:40:35] this may take a while depending on the size of your logs 
INFO[02-10|21:40:35] getting flynn-host logs 
INFO[02-10|21:40:35] getting sirenia metadata 
INFO[02-10|21:40:35] getting scheduler state 
INFO[02-10|21:40:35] getting job logs 
INFO[02-10|21:40:49] getting system information 
EROR[02-10|21:40:49] error capturing output of "iptables -L -v -n --line-numbers" err="exit status 3"
INFO[02-10|21:40:50] creating anonymous gist 
87.28 MB / 87.28 MB [========================================================================================================================================] 100.00 % 16.14 MB/s 0sEROR[02-10|21:41:05] unexpected HTTP status: 405 
87.28 MB / 87.28 MB [========================================================================================================================================] 100.00 % 5.62 MB/s 15s
21:41:05.695644 host.go:163: unexpected HTTP status: 405
```

Also of note is that one of my servers also appeared to have crashed & rebooted 26 hours ago.

**Comment by titanous** [comment]
@kenaniah Please collect with the `--tarball` option and upload it somewhere or email to logs@flynn.io.

**Comment by kenaniah** [comment]
URL sent via email. Would be really nice if the collection tool just worked. Even better if it just uploaded to flynn.io privately and automatically. Kinda frustrated that this is my 10th (!!) cluster I've built in the last 2 weeks.

**Comment by titanous** [comment]
@kenaniah You said "exact same issue", but can you please describe exactly what happened and any errors that you saw? Thanks.

**Comment by titanous** [comment]
@kenaniah Also, what time (UTC) did the issue occur?

**Comment by kenaniah** [comment]
All deploys now get stuck in slugbuilder at the same spot:
```
-----> Discovering process types
       Procfile declares types -> web, sidekiq
       Default process types for Multipack -> rake, console, web, worker
```

All `flynn scale` commands work, but all `flynn scale web=...` commands fail. 

I'm guessing it's some sort of issue in the controller. 

Last good build was at 2017-02-10T01:53:53Z. 
Issue was noticed at 2017-02-10T9:15:00Z (just over an hour ago)

**Comment by dottodot** [comment]
@kenaniah The only way round this I found was to scale the app you're deploying down to 1 and I've not had any issues since doing it that way. It seems that whatever the bug is that if you have your app scaled so it's running on all the nodes is completely breaks it, and like you I had to rebuild  the cluster multiple time before discovering this. I obviously can't guarantee it will work in your case, but might be worth trying until this get fixed.
